library/src/main/java/com/novoda/downloadmanager/lib/DownloadInfo.java

```diff
@@ -440,12 +440,13 @@ private NetworkState checkSizeAllowedForNetwork(int networkType) {
      * create a {DownloadThread} and enqueue it into given
      * {@link Executor}.
      *
-     * @param collatedDownloadInfo
      * @return If actively downloading.
      */
     public boolean isReadyToDownload(CollatedDownloadInfo collatedDownloadInfo) {
         synchronized (this) {
-            return isClientReadyToDownload(collatedDownloadInfo) && isDownloadManagerReadyToDownload();
+            // This order MATTERS
+            // it means completed downloads will not be accounted for in later downloadInfo queries
+            return isDownloadManagerReadyToDownload() && isClientReadyToDownload(collatedDownloadInfo);
```

Comment:
@Electryc The thing is that in the client checking mechanism, if it fails it will trigger the UI for displaying the errors. 

This fix prevents triggering them for already completed downloads so the client check is not invoked.

@blundell correct me if I'm wrong